superset-frontend/src/components/Chart/DrillBy/DrillByMenuItems.tsx

```diff
@@ -119,51 +157,59 @@ export const DrillByMenuItems = ({
   const verboseMap = useVerboseMap(dataset);
 
   useEffect(() => {
+    async function loadOptions() {
+      const datasetId = +formData.datasource.split('__')[0];
```

Comment:
While this method seems convenient, it might cause misunderstandings for those not familiar with JavaScript. Therefore, it would be better to apply a more explicit number conversion function.

```suggestion
      const datasetId = Number(formData.datasource.split('__')[0]);
```